fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/LoanSchedulePeriodData.java

```diff
@@ -402,6 +403,23 @@ private BigDecimal defaultToZeroIfNull(final BigDecimal possibleNullValue) {
         return value;
     }
 
+    public BigDecimal getRatePerDay() {
+        return getInterestOriginalDue().divide(BigDecimal.valueOf(getDaysInPeriod()), MoneyHelper.getMathContext());
+    }
+
+    public boolean isActualPeriod(final LocalDate businessDate) {
```

Comment:
Misleading hence `actual period` should include the due date, but now we need a logic which does not include it. Would you mind to rename it for something more specific? Also when it is on due date, the payable not due logic should be zero....

Reply:
Done! Renamed